src/explorer/DeploymentSlotsTreeItem.ts

```diff
@@ -94,6 +105,52 @@ export class DeploymentSlotsTreeItem implements IAzureParentTreeItem {
             }
         });
     }
+
+    private async chooseConfigurationSource(node: IAzureParentNode<DeploymentSlotsTreeItem>): Promise<IAzureNode<SiteTreeItem> | undefined> {
+        const deploymentSlots: IAzureNode[] = <IAzureNode[]>await node.getCachedChildren();
+        const configurationSources: util.IQuickPickItemWithData<IAzureNode | undefined>[] = [{
+            label: "Don't clone configuration from an existing slot",
+            description: '',
+            data: undefined
+        }];
+        // add the production slot itself
+        configurationSources.push({
+            label: (<SiteTreeItem>node.parent.treeItem).siteWrapper.appName,
+            description: '',
+            detail: '',
+            data: node.parent
+        });
+
+        // add the web app's current deployment slots
+        for (const slot of deploymentSlots) {
+            configurationSources.push({
+                label: (<SiteTreeItem>slot.treeItem).siteWrapper.appName,
+                description: '',
+                data: slot
+            });
+        }
+
+        const quickPickOptions = { placeHolder: `Choose a configuration source.`, ignoreFocusOut: true };
+        const result = await window.showQuickPick(configurationSources, quickPickOptions);
+        if (!result) {
+            throw new UserCancelledError();
+        }
+        return <IAzureNode<SiteTreeItem>>result.data;
+    }
+
+    private async parseAppSettings(node: IAzureNode<SiteTreeItem>): Promise<NameValuePair[]> {
+        const client = nodeUtils.getWebSiteClient(node);
+        const appSettings = node.treeItem.siteWrapper.slotName ?
+            await client.webApps.listApplicationSettingsSlot(node.treeItem.siteWrapper.resourceGroup, node.treeItem.siteWrapper.name, node.treeItem.siteWrapper.slotName) :
+            await client.webApps.listApplicationSettings(node.treeItem.siteWrapper.resourceGroup, node.treeItem.siteWrapper.name);
+        const appSettingPairs: NameValuePair[] = [];
+        // iterate String Directionary to parse into NameValuePair[]

```

Comment:
Directionary 😂

Reply:
Thank the Lord that Jing didn't see this one...